### PR TITLE
fix: After changing language and then pressing back button in UpdatePasswordFragment goes to home screen

### LINKED ISSUE #2075: After changing language and then pressing back button in UpdatePasswordFragment goes to home screen
**Summary:**

When Change password is clicked just after changing language and back button is pressed then it goes to home activity instead of the settings fragment.

**Steps to reproduce:**

- Go to Settings
- Change language
- Go to Change Password just after
- Press the back button

**Expected behaviour:**

Goes to Settings Fragment

**Observed behaviour:**

Goes to Home Activity

**Device and Android version:**

Poco X4 Pro 5G
Android 13 (Custom ROM)

**Screenshots:**

https://user-images.githubusercontent.com/85404940/227909529-08cb8382-a9b8-42d4-ab4e-ee493a147447.mp4

### PR DESCRIPTION
**Pull request by kingbass01**
Fixes #2075 

Please Add Screenshots If there are any UI changes.

https://user-images.githubusercontent.com/85404940/227917384-5ea82aea-ce51-4bab-91b0-4f2aecf03e96.mp4

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by kingbass01** [comment]
Hey @jawidMuhammadi, please review.

File: app/src/main/java/org/mifos/mobile/ui/activities/SettingsActivity.kt [hunk 1]
```diff
@@ -26,11 +26,16 @@ class SettingsActivity : BaseActivity() {
     }
 
     override fun onBackPressed() {
-        super.onBackPressed()
-        if (hasSettingsChanged) {
-            ActivityCompat.finishAffinity(this)
-            val i = Intent(this, HomeActivity::class.java)
-            startActivity(i)
+        val count = supportFragmentManager.backStackEntryCount
+        if (count == 0) {
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/ui/activities/SettingsActivity.kt]: What do you think about decreasing the number of lines by doing the following?

- `if (supportFragmentManager.backStackEntryCount == 0) {...}`
- `startActivity(Intent(this, HomeActivity::class.java))`

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/activities/SettingsActivity.kt (hunk 1 above)
Comment (kingbass01) [reply app/src/main/java/org/mifos/mobile/ui/activities/SettingsActivity.kt]: Yes I agree, no point in creating new variables. Changes are done.

**Review by jawidMuhammadi — APPROVED** [review]
LGTM! 👏

**Comment by gururani-abhishek** [comment]
Hi @jawidMuhammadi, I wanted to bring to your attention a small issue(#2089) that was introduced with this commit. I have fixed it with the commit #2093, which you can review whenever you have some spare time :)

**Comment by kingbass01** [comment]
Ah, sorry, my bad

**Comment by gururani-abhishek** [comment]
Absolutely no worries @kingbass01, keep contributing. Together we grow :)
I got to learn something really cool while fixing it, so my win.

**Pull request merged**